baseplate/server.go

```diff
@@ -0,0 +1,71 @@
+package baseplate
+
+import (
+	"errors"
+	"io/ioutil"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ServerConfig is a general purpose config for assembling a BaseplateServer
+type ServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Log struct {
+		Level log.Level
+	}
+
+	Metrics struct {
+		Namespace string
+		Endpoint  string
+	}
+
+	Secrets struct {
+		Path string
+	}
+
+	Sentry struct {
+		DSN         string
+		Environment string
+		SampleRate  float64
+	}
+
+	Tracing struct {
+		Namespace     string
+		Endpoint      string
+		RecordTimeout time.Duration
+		SampleRate    float64
+	}
+}
+
+// Server is the primary interface for baseplate servers.
+type Server interface {
+	Config() ServerConfig
+	Serve() error
+	Stop() error
```

Comment:
Also, why not make this `Close() error` so it's compatible with `io.Closer` as well? Since `Server` is an interface you can just embed `io.Closer` here.